### PR TITLE
refactor: NNBD docs migration

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

NNBD migration documentation for each package.

## Related Issues

[Internal ticket](https://yt.invertase.io/youtrack/issue/FLUTTERFIRE-34)
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by Salakar** [comment]
@googlebot I consent.

**Comment by Ehesp** [comment]
@googlebot I consent.

File: docs/null-safety.mdx [hunk 1]
```diff
@@ -12,24 +12,27 @@ The below table outlines the latest null-safe plugin versions:
...
 
 ---
 
 ### Using with other non-null-safe FlutterFire packages
 
-:::caution
-Mixing currently **published** null-safety FlutterFire packages alongside **published** non-nullsafety packages is not supported.
-:::
+The below table outlines the latest FlutterFire packages that **are not** null-safe, but are still compatible with other null-safe packages.
+(i.e. The application will build, but you lose null-safety).
```
Comment (Salakar) [inline docs/null-safety.mdx:27]:
```suggestion
(i.e. The application will build, but you lose sound null-safety).
```

File: docs/analytics/overview.mdx [hunk 2]
```diff
@@ -12,8 +12,63 @@ Analytics reports help you understand clearly how your users behave, which enabl
...
+    { label: "Legacy", value: "legacy" },
+    { label: "Null safety", value: "null-safe" },
+]}
+>
+<TabItem value="legacy">
+</TabItem>
+<TabItem value="null-safe">
+
+This package version is not null safe migrated yet, but has been implemented to be compatible with other null safe FlutterFire packages.
+With that in mind, you will need to use the `beta` channel which supports null safe projects.
```
Comment (Salakar) [inline docs/analytics/overview.mdx:28]:
```suggestion
:::caution
This package version is not null safe migrated yet, but has been implemented to be compatible with other null safe FlutterFire packages.
With that in mind, you will need to use the `beta` channel which supports null safe projects.
:::
```

File: docs/database/overview.mdx [hunk 3]
```diff
@@ -13,8 +13,63 @@ all of your clients can share one Realtime Database instance and automatically r
...
+    { label: "Legacy", value: "legacy" },
+    { label: "Null safety", value: "null-safe" },
+]}
+>
+<TabItem value="legacy">
+</TabItem>
+<TabItem value="null-safe">
+
+This package version is not null safe migrated yet, but has been implemented to be compatible with other null safe FlutterFire packages.
+With that in mind, you will need to use the `beta` channel which supports null safe projects.
```
Comment (Salakar) [inline docs/database/overview.mdx:29]:
```suggestion
:::caution
This package version is not null safe migrated yet, but has been implemented to be compatible with other null safe FlutterFire packages.
With that in mind, you will need to use the `beta` channel which supports null safe projects.
:::
```

File: docs/performance/overview.mdx [hunk 4]
```diff
@@ -11,8 +11,64 @@ Firebase Performance Monitoring is a service that helps you to gain insight into
...
+    { label: "Legacy", value: "legacy" },
+    { label: "Null safety", value: "null-safe" },
+]}
+>
+<TabItem value="legacy">
+</TabItem>
+<TabItem value="null-safe">
+
+This package version is not null safe migrated yet, but has been implemented to be compatible with other null safe FlutterFire packages.
+With that in mind, you will need to use the `beta` channel which supports null safe projects.
```
Comment (Salakar) [inline docs/performance/overview.mdx:27]:
```suggestion
:::caution
This package version is not null safe migrated yet, but has been implemented to be compatible with other null safe FlutterFire packages.
With that in mind, you will need to use the `beta` channel which supports null safe projects.
:::
```

File: docs/remote-config/overview.mdx [hunk 5]
```diff
@@ -15,8 +15,63 @@ apply them with a negligible impact on performance.
...
+    { label: "Legacy", value: "legacy" },
+    { label: "Null safety", value: "null-safe" },
+]}
+>
+<TabItem value="legacy">
+</TabItem>
+<TabItem value="null-safe">
+
+This package version is not null safe migrated yet, but has been implemented to be compatible with other null safe FlutterFire packages.
+With that in mind, you will need to use the `beta` channel which supports null safe projects.
```
Comment (Salakar) [inline docs/remote-config/overview.mdx:31]:
```suggestion
:::caution
This package version is not null safe migrated yet, but has been implemented to be compatible with other null safe FlutterFire packages.
With that in mind, you will need to use the `beta` channel which supports null safe projects.
:::
```

**Review by Salakar — APPROVED** [review]

**Review by rrousselGit — APPROVED** [review]

**Pull request merged**